test/scripting/stable_save.lua

```diff
@@ -1,13 +1,30 @@
 include "scripting/coroutine.lua"
 
 -- Save the game so that reloading does not skip
-function stable_save(game, savename)
-   local old_speed = game.desired_speed
+function stable_save(game, savename, desired_speed)
+   local mapview = wl.ui.MapView()
+   local last_save_time = game.last_save_time
+   print("###### stable_save: last save time " .. last_save_time)
+
    game.desired_speed = 1000
    sleep(100)
    game:save(savename)
    game.desired_speed = 1000
-   sleep(2000)  -- Give the loaded game a chance to catch up
-   game.desired_speed = old_speed
-   sleep(1000)
+
+   -- Wait until save was finished
+   local counter = 0
+   repeat
+      sleep(200)
```

Comment:
Forgotten counter increase.
Also what is the point of breaking after 20 iterations – it looks to me that breaking early would be a bug